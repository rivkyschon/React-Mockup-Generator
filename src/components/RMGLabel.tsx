import React from 'react';

interface RMGLabelProps {
  content: string;
}

const RMGLabel: React.FC<RMGLabelProps> = ({content}) => {
  return (
    <label
      className="block text-lg font-medium text-gray-700 mb-2"
      htmlFor="email">
      {content}
    </label>
  );
};

export default RMGLabel;
