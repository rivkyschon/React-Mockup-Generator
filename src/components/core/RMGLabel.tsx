import React from 'react';

interface RMGLabelProps {
  content: string;
  className?: string;
}

const RMGLabel: React.FC<RMGLabelProps> = ({content, className}) => {
  return (
    <label className={className || 'block text-lg font-medium text-gray-900'}>
      {content}
    </label>
  );
};

export default RMGLabel;
