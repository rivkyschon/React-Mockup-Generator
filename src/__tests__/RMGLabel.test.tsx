// src/components/__tests__/RMGLabel.test.tsx

import React from 'react';
import {render} from '@testing-library/react';
import RMGLabel from '../components/core/RMGLabel';

describe('RMGLabel', () => {
  it('renders correctly', () => {
    const {asFragment} = render(<RMGLabel content="Label Content" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct content', () => {
    const {getByText} = render(<RMGLabel content="Label Content" />);
    // expect(getByText('Label Content')).toBeInTheDocument();
  });
});
