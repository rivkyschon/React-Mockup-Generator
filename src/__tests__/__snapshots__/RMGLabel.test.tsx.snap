// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RMGLabel renders correctly 1`] = `
<DocumentFragment>
  <label
    class="block text-lg font-medium text-gray-700 mb-2"
    for="email"
  >
    Label Content
  </label>
</DocumentFragment>
`;
